dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/NettyFrameworkImpl.java

```diff
@@ -0,0 +1,359 @@
+    Set<Channel> activeChannels = ConcurrentHashMap.newKeySet();
+
+    protected EventLoopGroup parentGroup;
+    protected EventLoopGroup childGroup;
+    
+    private CHFWBundle chfw;
+    private volatile boolean isActive = false;
+
+    
+    /**
+     * DS method for setting the required channel framework service.
+     * TODO: for now this reference is needed for access to EndPointMgr. That code will be split out.
+     *
+     * @param bundle
+     */
+    @Reference(name = "chfwBundle")
+    protected void setChfwBundle(CHFWBundle bundle) {
+        chfw = bundle;
+    }
+
+    /**
+     * This is a required static reference, this won't
+     * be called until the component has been deactivated
+     *
+     * @param bundle CHFWBundle instance to unset
+     */
+    protected void unsetChfwBundle(CHFWBundle bundle) {}
+
+    @Activate
+    protected void activate(ComponentContext context, Map<String, Object> config) {
+        // use the executor service provided by Liberty
+        parentGroup = new NioEventLoopGroup(1, executorService);
+        childGroup = new NioEventLoopGroup(0, executorService);
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context, Map<String, Object> properties) {
+        serverStopping();
+    }
+
+    @Modified
+    protected void modified(ComponentContext context, Map<String, Object> config) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(this, tc, "Processing config", config);
+        }
+        // TODO: update any framework-specific config
+    }
+    
+    /**
+     * DS method for setting the executor service reference.
+     *
+     * @param executorService the {@link java.util.concurrent.ExecutorService} to
+     *                            queue work to.
+     */
+    @Reference(service = ExecutorService.class,
+               cardinality = ReferenceCardinality.MANDATORY)
+    protected void setExecutorService(ExecutorService executorService) {
+        this.executorService = executorService;
+    }
+
+    /**
+     * DS method for clearing the executor service reference.
+     * This is a required reference, will be called after deactivate.
+     *
+     * @param executorService the service instance to clear
+     */
+    protected void unsetExecutorService(ExecutorService executorService) {
+        this.executorService = null;
+    }
+
+    /**
+     * When notified that the server is going to stop, pre-quiesce all chains in the runtime.
+     * This will be called before services start getting torn down..
+     *
+     * @see com.ibm.wsspi.kernel.service.utils.ServerQuiesceListener#serverStopping()
+     */
+    @Override
+    public void serverStopping() {
+        if (isActive) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Destroying all endpoints");
+            }
+            isActive = false;
```

Comment:
need to be careful as this flag is used in an if statement in the stop(channel) method and setting it false will noop that method (if my reading is correct)